projects/packages/my-jetpack/_inc/components/product-interstitial/index.jsx

```diff
@@ -119,50 +122,48 @@ export default function ProductInterstitial( {
 				return;
 			}
 
-			activate()
-				// After product activation, we'll re-check the post-checkout redirect
-				// If the product is active, it may (very likely is) different
-				.then( activatedProduct => {
-					postCheckoutUrl = activatedProduct?.post_checkout_url
-						? activatedProduct.post_checkout_url
-						: myJetpackCheckoutUri;
-				} )
-				.finally( () => {
-					const postActivationUrl = product?.postActivationUrl;
-					const hasRequiredPlan = tier
-						? product?.hasRequiredTier?.[ tier ]
-						: product?.hasRequiredPlan;
-					const isFree = tier
-						? product?.pricingForUi?.tiers?.[ tier ]?.isFree
-						: product?.pricingForUi?.isFree;
-					const needsPurchase = ! isFree && ! hasRequiredPlan;
-
-					// If the product is CRM, redirect the user to the Jetpack CRM pricing page.
-					// This is done because CRM is not part of the WP billing system
-					// and we can't send them to checkout like we can with the rest of the products
-					if ( product.pluginSlug === 'zero-bs-crm' && ! hasRequiredPlan ) {
-						window.location.href = 'https://jetpackcrm.com/pricing/';
-						return;
-					}
+			activate(
+				{ productId: slug },
+				{
+					onSettled: ( { productId: activatedProduct } ) => {
+						postCheckoutUrl = activatedProduct?.post_checkout_url
+							? activatedProduct.post_checkout_url
+							: myJetpackCheckoutUri;
+						const postActivationUrl = product?.postActivationUrl;
+						const hasRequiredPlan = tier
+							? product?.hasRequiredTier?.[ tier ]
+							: product?.hasRequiredPlan;
+						const isFree = tier
+							? product?.pricingForUi?.tiers?.[ tier ]?.isFree
+							: product?.pricingForUi?.isFree;
+						const needsPurchase = ! isFree && ! hasRequiredPlan;
 
-					// If no purchase is needed, redirect the user to the product screen.
-					if ( ! needsPurchase ) {
-						if ( postActivationUrl ) {
-							window.location.href = postActivationUrl;
+						// If the product is CRM, redirect the user to the Jetpack CRM pricing page.
+						// This is done because CRM is not part of the WP billing system
+						// and we can't send them to checkout like we can with the rest of the products
+						if ( product.pluginSlug === 'zero-bs-crm' && ! hasRequiredPlan ) {
+							window.location.href = 'https://jetpackcrm.com/pricing/';
```

Comment:
Any links I would put in const instead of hardcoding.

Reply:
This was here before this patch, I think we can leave this for a new PR